controllers/drplacementcontrol.go

```diff
@@ -877,6 +914,35 @@ func (d *DRPCInstance) updateCondition(condition *metav1.Condition, idx int, rea
 	d.needStatusUpdate = true
 }
 
+func (d *DRPCInstance) namespaceExistsOnManagedCluster(cluster string) (bool, error) {
+	exists := true
+
+	// create ManagedClusterView to check if namespace exists
+	namespace, err := d.reconciler.MCVGetter.GetNamespaceFromManagedCluster(d.instance.Name, cluster, d.instance.Namespace)
+	if err != nil {
+		if errors.IsNotFound(err) { // successfully detected that Namespace is not found by ManagedClusterView
+			d.log.Info(fmt.Sprintf("Namespace '%s' not found on cluster %s", d.instance.Namespace, cluster))
+
+			return !exists, nil
+		}
+
+		d.log.Error(err, "failed get Namespace from ManagedCluster")
+
+		return !exists, errorswrapper.Wrap(err, "failed to get Namespace from managedcluster")
+	}
+
+	// sanity checks: only allow Active namespaces
+	if namespace == nil {
```

Comment:
(nit) I would remove this extra code.  No reason to double test.  If no error, we just have to trust that we have a namespace object and if not, we will panic below anyway.